Civi/Token/TokenRow.php

```diff
@@ -136,13 +136,13 @@ public function tokens($a = NULL, $b = NULL, $c = NULL) {
    */
   public function customToken($entity, $customFieldID, $entityID) {
     $customFieldName = "custom_" . $customFieldID;
-    $fieldValue = civicrm_api3($entity, 'getvalue', array(
-      'return' => $customFieldName,
-      'id' => $entityID,
+    $record = civicrm_api3($entity, "getSingle", array(
```

Comment:
not having a return does impact the speed of the api - esp on contribution.get -